Libraries/src/Amazon.Lambda.RuntimeSupport/Helpers/Logging/AbstractLogMessageFormatter.cs

```diff
@@ -0,0 +1,288 @@
+            for (int i = 0, l = messageTemplate.Length; i < l; i++)
+            {
+                var c = messageTemplate[i];
+
+                // If not using positional properties and we have hit the point there are more message properties then arguments
+                // then just add the rest of the message template onto the messageBuilder.
+                if (!usePositional && (messageProperties.Count <= propertyIndex || messageArguments?.Length <= propertyIndex))
+                {
+                    messageBuilder.Append(c);
+                    continue;
+                }
+
+                switch (c)
+                {
+                    case '{':
+                        if (state == LogFormatParserState.InMessage)
+                        {
+                            // regardless of whether this is the opening of a parameter we'd still need to add {
+                            propertyBuilder.Append(c);
+                            state = LogFormatParserState.PossibleParameterOpen;
+                        }
+                        else if (state == LogFormatParserState.PossibleParameterOpen)
+                        {
+                            // We have hit an escaped "{" by the user using "{{". Since we now know we are
+                            // not in a message properties write back the propertiesBuilder into 
+                            // messageBuilder and reset the propertyBuilder.
+                            messageBuilder.Append(propertyBuilder.ToString());
+                            propertyBuilder.Clear();
+                            messageBuilder.Append(c);
+                            state = LogFormatParserState.InMessage;
+                        }
+                        else
+                        {
+                            propertyBuilder.Append(c);
+                        }
+                        break;
+                    case '}':
+                        if (state == LogFormatParserState.InMessage)
+                        {
+                            messageBuilder.Append(c);
+                        }
+                        else
+                        {
+                            var property = messageProperties[propertyIndex];
+                            object argument = null;
+                            if(usePositional)
+                            {
+                                // If usePositional is true then we have confirmed the `Name` property is an int
+                                var index = int.Parse(property.Name, CultureInfo.InvariantCulture);
+                                if (index < messageArguments.Length)
+                                {
+                                    argument = messageArguments[index];
+                                }
+                            }
+                            else
+                            {
+                                argument = messageArguments[propertyIndex];
```

Comment:
Could you please add a null check for `messageArguments` before indexing into it? I got a warning for it in Rider.

Reply:
I added a short circuit at the top that exits out if `messageArguments` is null or empty.